beacon/goclient/goclient.go

```diff
@@ -174,11 +170,15 @@ func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.Op
 	}
 
 	// Get the node's version and client.
-	client.nodeVersion, err = client.client.NodeVersion(opt.Context)
+	nodeVersionResp, err := client.client.NodeVersion(opt.Context, &api.NodeVersionOpts{})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get node version")
+		return nil, fmt.Errorf("failed to get node version: %w", err)
+	}
+	if nodeVersionResp == nil {
+		return nil, fmt.Errorf("node version response is nil")
 	}
-	client.nodeClient = ParseNodeClient(client.nodeVersion)
+	client.nodeVersion = nodeVersionResp.Data
```

Comment:
at other places you check if `Data` is `nil` why not here?

Reply:
the Data is string in this case